### PR TITLE
feat: Add `path` parameter to Litestar application class

### PR DESCRIPTION
**Pull request by kedod**
## Description

- Expose `path` parameter at `Litestar` application class level

At this moment if we want use prefixed path for all routes in the application we have to create 
dedicated router. E.g:
```python
from litestar import Litestar, Router, get


@get("/")
async def index() -> str:
    return "Hello, world!"


router = Router(path="/path", route_handlers=[index])


app = Litestar(route_handlers=[router])
```

This PR allows to set `path` on the `Litestar` application level.
Thanks to this we don't need additional `Router` class.

```python
from litestar import Litestar, get


@get("/")
async def index() -> str:
    return "Hello, world!"


app = Litestar(path="/path", route_handlers=[index])
```

### CODE REVIEW

**Review by cofin — APPROVED** [review]
LGTM

File: litestar/app.py [hunk 1]
```diff
@@ -277,6 +278,8 @@ def __init__(
                 :class:`ASGI Scope <.types.Scope>`.
             parameters: A mapping of :class:`Parameter <.params.Parameter>` definitions available to all application
                 paths.
+            path: A path fragment that is prefixed to all route handlers, controllers and routers associated
+                with the application instance.
```
Comment (JacobCoffee) [inline litestar/app.py:282]:
```suggestion
            path: A path fragment that is prefixed to all route handlers, controllers and routers associated
                with the application instance.
                
                .. versionadded:: 2.8.0
```

File: litestar/testing/helpers.py [hunk 2]
```diff
@@ -201,6 +202,8 @@ def test_my_handler() -> None:
             :class:`ASGI Scope <.types.Scope>`.
         parameters: A mapping of :class:`Parameter <.params.Parameter>` definitions available to all application
             paths.
+        path: A path fragment that is prefixed to all route handlers, controllers and routers associated
+            with the application instance.
```
Comment (JacobCoffee) [inline litestar/testing/helpers.py:206]:
```suggestion
        path: A path fragment that is prefixed to all route handlers, controllers and routers associated
            with the application instance.
            
            .. versionadded:: 2.8.0
```

File: litestar/testing/helpers.py [hunk 3]
```diff
@@ -456,6 +461,8 @@ async def test_my_handler() -> None:
...
```
Comment (JacobCoffee) [inline litestar/testing/helpers.py:465]:
```suggestion
        path: A path fragment that is prefixed to all route handlers, controllers and routers associated
            with the application instance.
            
            .. versionadded:: 2.8.0
```

File: litestar/config/app.py [hunk 4]
```diff
@@ -151,6 +151,7 @@ class AppConfig:
     """
     parameters: ParametersMap = field(default_factory=dict)
     """A mapping of :class:`Parameter <.params.Parameter>` definitions available to all application paths."""
+    path: str = field(default="")
```
Comment (JacobCoffee) [inline litestar/config/app.py:154]:
```suggestion
    path: str = field(default="")
    """A base path that prefixed to all route handlers, controllers and routers associated with the 
    application instance.
    
    .. versionadded:: 2.8.0
    """
```

**Review by JacobCoffee — CHANGES_REQUESTED** [review]

**Comment by JacobCoffee** [comment]
seems good :) thx

**Pull request merged**